application/scraper.go

```diff
@@ -0,0 +1,223 @@
+			selector.Scope = scrapeConfig.ID.String()
+			return selector
+		})
+
+		specJSON, err := json.Marshal(spec)
+		if err != nil {
+			return err
+		}
+
+		// generate a deterministic id for the scraper based on the application id
+		id := uuid.UUID(uuidV5.NewV5(uuidV5.UUID(scrapeConfig.ID), string(app.UID)))
+		scraper := &models.ConfigScraper{
+			ID:            id,
+			Namespace:     scrapeConfig.Namespace,
+			Name:          fmt.Sprintf("%s/%s-%s-app-generated", scrapeConfig.Namespace, app.Name, scrapeConfig.Name),
+			Description:   scrapeConfig.Description,
+			Spec:          string(specJSON),
+			Source:        models.SourceApplicationCRD,
+			ApplicationID: lo.ToPtr(app.GetID()),
+		}
+		if err := ctx.DB().Save(scraper).Error; err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func linkToConfigs(ctx context.Context, app *v1.Application) error {
+	configIDs, err := query.FindConfigIDsByResourceSelector(ctx, -1, app.Spec.Mapping.Logins...)
+	if err != nil {
+		return err
+	}
+
+	relationships := make([]models.ConfigRelationship, len(configIDs))
+	for i, configID := range configIDs {
+		relationships[i] = models.ConfigRelationship{
+			ConfigID:  string(app.GetUID()),
+			RelatedID: configID.String(),
+			Relation:  "ApplicationLogin",
+		}
+	}
+
+	if err := ctx.DB().Save(relationships).Error; err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func SyncApplicationScrapeConfigs(sc context.Context) *job.Job {
+	return &job.Job{
+		Name:          "ApplicationConfigScraperSync",
+		Context:       sc,
+		Schedule:      "@every 10m",
+		Singleton:     true,
+		JitterDisable: true,
+		JobHistory:    true,
+		Retention:     job.RetentionFew,
+		RunNow:        true,
+		Fn: func(jr job.JobRuntime) error {
+			applications, err := db.GetAllApplications(sc)
+			if err != nil {
+				return err
+			}
+
+			for _, application := range applications {
+				app, err := v1.ApplicationFromModel(application)
+				if err != nil {
+					return err
+				}
+
+				if err := generateConfigScraper(sc, app); err != nil {
+					return err
+				}
+
+				if err := linkToConfigs(sc, app); err != nil {
+					return err
+				}
+
+				// TODO: Decide on this
+				// Maybe config access should be linked to application ID directly.
+				scraperID := uuid.MustParse("6e1fe094-550f-40c7-9286-d4e4fafebdcc")
+				if err := generateCustomRoles(sc, scraperID, app); err != nil {
+					return err
+				}
+			}
+
+			applicationIDs := lo.Map(applications, func(app models.Application, _ int) uuid.UUID {
+				return app.ID
+			})
+
+			return cleanupStaleScrapers(sc, applicationIDs)
+		},
+	}
+}
+
+// Generate new custom roles & config accesses for those roles
+func generateCustomRoles(ctx context.Context, scraperID uuid.UUID, app *v1.Application) error {
+	for _, role := range app.Spec.Mapping.Roles {
+		roleID := uuid.UUID(uuidV5.NewV5(uuidV5.NamespaceDNS, fmt.Sprintf("%s-%s", app.UID, role.Role)))
+
+		externalRole := models.ExternalRole{
```

Comment:
This should link to applicationId, not scraper

Reply:
https://github.com/flanksource/duty/pull/1443